storage/txcache/txByHashMap.go

```diff
@@ -34,19 +34,18 @@ func (txMap *txByHashMap) addTx(tx *WrappedTransaction) bool {
 
 // removeTx removes a transaction from the map
 func (txMap *txByHashMap) removeTx(txHash string) (*WrappedTransaction, bool) {
-	tx, ok := txMap.getTx(txHash)
-	if !ok {
+	item, removed := txMap.backingMap.Remove(txHash)
+	if !removed {
 		return nil, false
 	}
 
-	// TODO / bugfix: when we concurrently try to remove the same transaction,
-	// we might end up decrementing the counters twice.
-	// Possible solution: use an "onItemRemoved" callback in the "backingMap"
-	// to decrement the counters.
+	tx := item.(*WrappedTransaction)
```

Comment:
You can check for correct type assertion here, so if will be the case in some strange circumstances you will avoid passing nil to line 46 in estimateTxSize method and also to return nil, true in line 49

```
tx, ok := item.(*WrappedTransaction)
if !ok {
   return nil, false
}
```